libraries/tock-register-interface/src/registers.rs

```diff
@@ -250,6 +262,55 @@ impl<T: IntLike, R: RegisterLongName> WriteOnly<T, R> {
     }
 }
 
+impl<T: IntLike, R: RegisterLongName, W: RegisterLongName> RWAlias<T, R, W> {
```

Comment:
Given the code duplication, I'm wondering whether it would make more sense to define *traits* for reading and writing registers.

Then `RWAlias` would be defined as follows (if that makes sense).

```rust
trait WriteRegister<T: IntLike, W: RegisterLongName> {
    fn set(&self, value: T);

    fn write(&self, field: FieldValue<T, R>) {
        self.set(field.value);
    }
}

impl<T: IntLike, R: RegisterLongName, W: RegisterLongName> WriteRegister<T, W> for RWAlias<T, R, W> {
    fn set(&self, value: T) {
        unsafe { ::core::ptr::write_volatile(&self.value as *const T as *mut T, value) }
    }

    // All other methods automatically derived.
}

impl<T: IntLike, R: RegisterLongName, W: RegisterLongName> ReadRegister<T, R> for RWAlias<T, R, W> {
    fn get(&self) -> T {
        unsafe { ::core::ptr::read_volatile(&self.value) }
    }

    // All other methods automatically derived.
}
```

The other types in this file could also be simplified with this pattern.

Reply:
I agree that the code duplication does not seem ideal.  Wouldn't the trait approach require changes to what consumers imported into their scope, though?  If that's the case, I'd rather it be done as follow-up work.

Reply:
Yes the scope could definitely be beyond this pull-request. And indeed having to import the traits manually would bring more overhead on users.

Another possibility would be to implement all the masking arithmetic as methods of the `Field<T, R>` type - there would not be less functions, but still less duplicated business logic.

Either way, it should probably not block this pull-request.